crates/corro-agent/src/agent/handlers.rs

```diff
@@ -273,7 +272,24 @@ pub async fn handle_notifications(
         trace!("handle notification");
         match notification {
             Notification::MemberUp(actor) => {
-                let member_added_res = agent.members().write().add_member(&actor);
+                // check for last empty when adding a new member.
+                let last_cleared_ts = {
+                    match agent.pool().read().await {
+                        Ok(conn) => match get_last_cleared_ts(&conn, actor.id()) {
+                            Ok(ts) => ts,
+                            Err(e) => {
+                                error!("could not get last_empty_ts: {e}");
+                                None
+                            }
+                        },
+                        Err(e) => {
+                            error!("could not get read conn: {e}");
+                            None
+                        }
+                    }
+                };
```

Comment:
It might be best to check the `member_added_res` and only query the DB if it is a newly added member (and then using `update_last_empty`).

The way SWIM works (or at least, `foca`), this can be called fairly often. I'll admit I don't remember if it's often enough to require moving this logic around, but I feel like changing it would be more consistent with how we usually manage the members list.